### PR TITLE
DashPay: update firebase (NMA-1064)

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

Updating Firebase BOM to 28.4.2 also requires updating DashJ to use protobuf-javalite 3.14.0
## Related PR's and Dependencies

https://github.com/dashevo/dashj/pull/137
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/AndroidManifest.xml [hunk 1]
```diff
@@ -96,7 +96,6 @@
             android:configChanges="keyboard|keyboardHidden"
             android:exported="true"
             android:label="@string/app_name_short"
-            android:launchMode="singleTask"
```
Comment (Syn-McJ) [inline wallet/AndroidManifest.xml:99]: Was there a problem with this solution?

**Review by Syn-McJ — APPROVED** [review]
Looks good

File: wallet/AndroidManifest.xml (hunk 1 above)
Comment (HashEngineering) [reply wallet/AndroidManifest.xml:99]: Yes, it resulted in multiple tasks or duplicate instances on the recent app lists when the app was started.

To solve the original issue, we also need to specify theses flags along with removing `singleTask`:
```
intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TOP)
```

This is one of the side effects of using `taskAffinity=""` which prevents a particular vulnerability called Strandhogg v1.  Patching up this vulnerability created many problems for us.

Comment (Syn-McJ) [reply wallet/AndroidManifest.xml:99]: Gotcha.

**Pull request merged**